MovieApp/MovieCategoriesListViewController.swift

```diff
@@ -85,62 +96,80 @@ class MovieCategoriesViewController: UIViewController {
-        whatsPopularLabel.attributedText = NSMutableAttributedString().bold("What's popular", fontSize: 20)
         
         popularCollectionView.autoPinEdge(.leading, to: .leading, of: whatsPopularLabel, withOffset: 0)
         popularCollectionView.autoPinEdge(.top, to: .bottom, of: whatsPopularLabel, withOffset: -15)
         popularCollectionView.autoSetDimension(.height, toSize: 220)
         popularCollectionView.autoPinEdge(.trailing, to: .trailing, of: view)
-        popularCollectionView.delegate = self
-        popularCollectionView.dataSource = popularCollectionViewDataSource
         
         freeToWatchLabel.autoMatch(.width, to: .width, of: whatsPopularLabel)
         freeToWatchLabel.autoMatch(.height, to: .height, of: whatsPopularLabel)
         freeToWatchLabel.autoPinEdge(.leading, to: .leading, of: whatsPopularLabel)
         freeToWatchLabel.autoPinEdge(.top, to: .bottom, of: popularCollectionView, withOffset: 25)
-        freeToWatchLabel.attributedText = NSMutableAttributedString().bold("Free to Watch", fontSize: 20)
         
         freeCollectionView.autoPinEdge(.leading, to: .leading, of: whatsPopularLabel, withOffset: 0)
         freeCollectionView.autoPinEdge(.top, to: .bottom, of: freeToWatchLabel, withOffset: -15)
         freeCollectionView.autoSetDimension(.height, toSize: 220)
         freeCollectionView.autoPinEdge(.trailing, to: .trailing, of: view)
-        freeCollectionView.delegate = self
-        freeCollectionView.dataSource = freeCollectionViewDataSource
         
         trendingLabel.autoMatch(.width, to: .width, of: whatsPopularLabel)
         trendingLabel.autoMatch(.height, to: .height, of: whatsPopularLabel)
         trendingLabel.autoPinEdge(.leading, to: .leading, of: whatsPopularLabel)
         trendingLabel.autoPinEdge(.top, to: .bottom, of: freeCollectionView, withOffset: 25)
-        trendingLabel.attributedText = NSMutableAttributedString().bold("Trending", fontSize: 20)
         
         trendingCollectionView.autoPinEdge(.leading, to: .leading, of: whatsPopularLabel, withOffset: 0)
         trendingCollectionView.autoPinEdge(.top, to: .bottom, of: trendingLabel, withOffset: -15)
         trendingCollectionView.autoSetDimension(.height, toSize: 220)
         trendingCollectionView.autoPinEdge(.trailing, to: .trailing, of: view)
-        trendingCollectionView.delegate = self
-        trendingCollectionView.dataSource = trendingCollectionViewDataSource
+        
+        // TODO
+        mainScrollView.contentSize = CGSize(width: view.bounds.width, height: trendingCollectionView.frame.maxY + 20)
         
     }
     
     private func registerCollectionViews() {
 
         for popularMovie in movieDetails.popularMovies{
-            popularCollectionView.register(MovieCategoriesViewCell.self, forCellWithReuseIdentifier: popularMovie.name)
+            popularCollectionView.register(MovieCategoriesListViewCell.self, forCellWithReuseIdentifier: popularMovie.name)
```

Comment:
This for part is really not needed. I saw that in the last homework and you did not correct it. You dont need to register a cell for each movie. You do it once and thats enough. Ie.

```
popularCollectionView.register(MovieCategoriesListViewCell.self, forCellWithReuseIdentifier: popularMovie.name)
freeCollectionView.register(MovieCategoriesListViewCell.self, forCellWithReuseIdentifier: freeMovie.name)
trendingCollectionView.register(MovieCategoriesListViewCell.self, forCellWithReuseIdentifier: trendingMovie.name)
```